admin-dev/themes/new-theme/js/pages/product/combination/image-selector.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ProductMap from '@pages/product/product-map';
+
+const {$} = window;
+
+export default class ImageSelector {
+  constructor() {
+    this.$selectorContainer = $(ProductMap.combinations.images.selectorContainer);
+    this.init();
+  }
+
+  init() {
+    $(ProductMap.combinations.images.checkboxContainer, this.$selectorContainer).hide();
+    this.$selectorContainer.on('click', ProductMap.combinations.images.imageChoice, (event) => {
+      const $imageChoice = $(event.currentTarget);
+      const $checkbox = $(ProductMap.combinations.images.checkbox, $imageChoice);
+
+      if ($checkbox.is(':checked')) {
+        $imageChoice.removeClass('selected');
+        $checkbox.prop('checked', false);
+      } else {
+        $imageChoice.addClass('selected');
+        $checkbox.prop('checked', true);
+      }
```

Comment:
```suggestion
      const isChecked = $checkbox.prop('checked');
      $imageChoice.toggleClass('selected', isChecked);
      $checkbox.prop('checked', !isChecked);
```

It will be a little faster because jquery doesn't have to parse and process a selector to figure out what to do